client/shared/src/main/scala/org/http4s/client/middleware/Metrics.scala

```diff
@@ -74,27 +85,29 @@ object Metrics {
   private def executeRequestAndRecordMetrics[F[_]](
       client: Client[F],
       ops: MetricsOps[F],
-      classifierF: Request[F] => Option[String],
+      classifierF: Request[F] => F[Option[String]],
       req: Request[F],
       statusRef: Ref[F, Option[Status]],
       start: Long
   )(implicit F: Clock[F], C: Concurrent[F]): Resource[F, Response[F]] =
     (for {
-      _ <- Resource.make(ops.increaseActiveRequests(classifierF(req)))(_ =>
-        ops.decreaseActiveRequests(classifierF(req)))
+      classifier <- Resource.eval(classifierF(req))
```

Comment:
What if this fails?  I guess we just won't record metrics.  Maybe this is the responsibility of the caller of `effect` to handle any errors and make a default `classifier`.  That's probably worth noting in the scaladoc.